core/base/src/TString.cxx

```diff
@@ -363,6 +379,10 @@ TString& TString::Append(char c, Ssiz_t rep)
 {
    if (!rep) return *this;
 
+   if (rep < 0) {
+      Error("TString::Append", "Negative length!");
+      Zero();
+   }
```

Comment:
```suggestion
   if (rep < 0) {
      Error("TString::Append", "Negative length!");
      return *this
   }
```
Since there length is negative, there is technically nothing to add and doing an "Append" should not result (even in fail mode) with the string being emptied.